### PR TITLE
Fix time range being incorrectly reported to start at 0

### PR DESCRIPTION
**Pull request by achim-k**
### Changelog
Fix time range being incorrectly reported to start at 0

### Docs
None

### Description
Fixes that the time range being incorrectly reported to start at 0 by using only data & log messages for determining the time range.

### CODE REVIEW

File: src/ULog.ts [hunk 1]
```diff
@@ -223,28 +223,35 @@ export class ULog {
...
           maxTimestamp = message.timestamp;
         }
         timeIndex.push([message.timestamp, i++, message]);
         logMessageCount++;
       } else {
-        timeIndex.push([maxTimestamp, i++, message]);
+        // Other message types do not have a timestamp field, so we use the current max. timestamp or
+        // alternatively the header timestamp which denotes when the logging started.
+        const timestamp = maxTimestamp ?? this._header.timestamp;
+        timeIndex.push([timestamp, i++, message]);
```
Comment (jtbandes) [inline src/ULog.ts]: Why do messages without a timestamp field need to be in the timeIndex?

Comment (achim-k) [reply src/ULog.ts]: Was wondering the same. The reason is, that the current `readMessages` generator function returns all messages in the data section. If we limit this to only return message types with timestamps, then we do not have to add messages without a timestamp to the time index.

Comment (jtbandes) [reply src/ULog.ts]: Are these other message types that don't have timestamps used for anything? Seems like we could either not emit them from readMessages or skip them when building the index?

Comment (achim-k) [reply src/ULog.ts]: > Are these other message types that don't have timestamps used for anything?

At least not by us. Not sure if there are any other users of this package (other than Lichtblick). 
~I think I will go ahead and not emit these kind of messages.~

Edit: I kept the types of emitted messages the same and only changed the time range calculation

File: src/ULog.test.ts [hunk 2]
```diff
@@ -301,7 +301,7 @@ describe("README.md", () => {
     const ulog = new ULog(reader);
     await ulog.open(); // required before any other operations
     expect(ulog.messageCount()).toBe(64599); // ex: 64599
-    expect(ulog.timeRange()).toEqual([0n, 181493506n]); // ex: [ 0n, 181493506n ]
+    expect(ulog.timeRange()).toEqual([112574307n, 181493506n]);
```
Comment (jtbandes) [inline src/ULog.test.ts]: magical - we already had a (broken) test case 🎉

File: src/ULog.ts [hunk 3]
```diff
@@ -214,30 +215,28 @@ export class ULog {
...
     const dataCounts = new Map<number, number>();
+    let minTimestamp: bigint | undefined;
     let maxTimestamp = 0n;
     let logMessageCount = 0;
 
     let i = 0;
     let message: DataSectionMessage | undefined;
     while ((message = await this.readParsedMessage())) {
       if (message.type === MessageType.Data) {
+        minTimestamp ??= message.value.timestamp;
```
Comment (jtbandes) [inline src/ULog.ts]: If we are never checking with `<`, how do we know this is the min?

Comment (achim-k) [reply src/ULog.ts]: arg, good catch! I forgot that these messages are not necessarily ordered

File: src/ULog.test.ts (hunk 2 above)
Comment (achim-k) [reply src/ULog.test.ts]: well, turned out that there is really a data message with 0n timestamp. Change here was related to the minTimestamp assignment bug

**Review by jtbandes — APPROVED** [review]

**Pull request merged**